swap_meet/vendor.py

```diff
@@ -1,2 +1,122 @@
+        """
+        Swaps this Vendor's my_item with the other Vendor's their_item. 
+        Returns True if swap is successful.
+        Returns False, if this Vendor's inventory doesn't contain my_item
+        or the other Vendor's inventory doesn't contain their_item.
+        """
+        if my_item in self.inventory and their_item in other.inventory:
+            return self.swap(other, my_item, their_item)
+        return False
+    
+    def swap_first_item(self, other):
+        """
+        Swaps this Vendor's first item with the other Vendor's first item.
+        Returns True if swap is successful.
+        Returns False, if either this Vendor or the other Vendor have an empty inventory.
+        """
+        if self.inventory and other.inventory:
+            return self.swap(other, self.get_item_by_index(0), other.get_item_by_index(0))
+        return False
+    
+    def get_best_by_category(self, category):
+        """
+        Returns the item with the highest condition and matching category from the inventory.
+        """
+        items_in_category = self.get_by_category(category)
+        if not items_in_category:
+            return None
+        return max(items_in_category, key = lambda item: item.condition)
+    
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        """
+        Swaps this Vendor's best item whose category matches their_priority with 
+        the other Vendor's best item whose category matches my_priority.
+        Returns True if swap is successful.
+        Returns False, if either this Vendor or the other Vendor have no item matching the given category.
+        """
+        my_item = self.get_best_by_category(their_priority)
+        their_item = other.get_best_by_category(my_priority)
+        if my_item and their_item:
+            return self.swap(other, my_item, their_item)
+        return False
+    
+    def get_newest(self):
+        """
+        Returns the item with the minimum age from the inventory.
+        Returns None if age is not specified.
+        """
+        newest_item, minimum_age = None, float('inf')
+        for item in self.inventory:
+            if item.age < minimum_age:
+                newest_item = item
+                minimum_age = item.age
+        return newest_item
+    
+    def swap_by_newest(self, other):
+        """
+        Swaps this Vendor's nwest item with the other Vendor's newest item.
+        Returns True if swap is successful.
+        Returns False, if either this Vendor or the other Vendor have an empty inventory.
+        """
+        my_item = self.get_newest()
+        their_item = other.get_newest()
+        if my_item and their_item:
+            return self.swap(other, my_item, their_item)
+        return False
+
+#======================================#        
+#========== Helper Functions ==========#
+#======================================#  
+    def get_item_by_index(self, index):
```

Comment:
This helper function is also not necessary. We do not actually want IndexError to happen in our code. The `swap_first_item` function already checks to see if the inventory is not empty before continuing. You can safely call `self.inventory[0]` in the `swap_first_item` function.

Reply:
I wanted to keep the program extensible by implementing this helper function. For example, we'd like to the swap the 7th items. Then, I can leverage this function to check if the 7th item is in the inventory or not.